### PR TITLE
executor: replace Lock operations in `MemTracker.Consume` with atomic operations to improve performance

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Lock operations in `MemTracker.Consume` can be replaced by atomic operations.

### What is changed and how it works?
Replace these Lock operations with atomic operation.

Before replace, the benchmark result is:
```
pkg: github.com/pingcap/tidb/util/memory
BenchmarkConsume-4      30000000                43.9 ns/op
```

After replace:
```
pkg: github.com/pingcap/tidb/util/memory
BenchmarkConsume-4      50000000                29.5 ns/op
```

### Check List 

Tests

### CODE REVIEW

File: util/memory/tracker.go [hunk 1]
```diff
@@ -136,12 +137,9 @@ func (t *Tracker) ReplaceChild(oldChild, newChild *Tracker) {
 func (t *Tracker) Consume(bytes int64) {
 	var rootExceed *Tracker
 	for tracker := t; tracker != nil; tracker = tracker.parent {
-		tracker.Lock()
-		tracker.bytesConsumed += bytes
-		if tracker.bytesLimit > 0 && tracker.bytesConsumed >= tracker.bytesLimit {
+		if atomic.AddInt64(&tracker.bytesConsumed, bytes) >= tracker.bytesLimit && tracker.bytesLimit > 0 {
```
Comment (lamxTyler) [inline util/memory/tracker.go]: Is it safe to mix the use of `atomic` and `lock`? Other places that read or write `bytesConsumed` still uses lock.

Comment (qw4990) [reply util/memory/tracker.go]: My fault, now all places using `bytesConsumed` has been replaced by atomic operations.

File: util/memory/tracker.go [hunk 2]
```diff
@@ -150,9 +148,7 @@ func (t *Tracker) Consume(bytes int64) {
 
 // BytesConsumed returns the consumed memory usage value in bytes.
 func (t *Tracker) BytesConsumed() int64 {
-	t.Lock()
-	defer t.Unlock()
-	return t.bytesConsumed
+	return atomic.LoadInt64(&t.bytesConsumed)
```
Comment (eurekaka) [inline util/memory/tracker.go]: Please also add comments in `Tracker` to make it clear that `Mutex` is only used for protecting `children` field now?

Comment (qw4990) [reply util/memory/tracker.go]: Sure, comments have been updated now.

File: util/memory/tracker.go [hunk 3]
```diff
@@ -37,14 +38,14 @@ import (
 // 1. Only "BytesConsumed()", "Consume()", "AttachTo()" and "Detach" are thread-safe.
 // 2. Other operations of a Tracker tree is not thread-safe.
 type Tracker struct {
-	sync.Mutex // For synchronization.
+	sync.Mutex            // It's used to protect children.
+	children   []*Tracker // The children memory trackers; protected by the Mutex above.
```
Comment (zz-jason) [inline util/memory/tracker.go]: how about:
```go
mu struct {
	sync.Mutex{}
	children []*Tracker
}
```

Comment (qw4990) [reply util/memory/tracker.go]: Okay, it's has been updated.

**Pull request merged**